CMakeLists.txt

```diff
@@ -253,6 +251,16 @@ add_compile_options(/MP) # Build with Multiple Processes (number of processes eq
 add_compile_options(/GS) # Buffer Security Check
 add_compile_options(/Zm200) # Specify Precompiled Header Memory Allocation Limit of 150MB
 add_compile_options(/wd4960 /wd4961 /wd4603 /wd4627 /wd4838 /wd4456 /wd4457 /wd4458 /wd4459 /wd4091 /we4640)
+add_compile_options(/Zi) # enable debugging information
+add_compile_options(/O1) # minimize space
+
+if (CMAKE_BUILD_TYPE STREQUAL Release OR CMAKE_BUILD_TYPE STREQUAL Relwithdebinfo)
+  add_compile_options(/GL) # enable WholeProgramOptimization
```

Comment:
It should be just "/GL" switch here. CoreCLR should be building with "minimize size" by default. It results into better overall performance. We selectively enable "optimize for speed" via pragmas in places where it is needed. And /Zi and /O1 are redundant because of they are specified above already.